plugins/API/API.php

```diff
@@ -195,6 +197,14 @@ public function getSegmentsMetadata($idSites = array(), $_hideImplementationData
 
         $cache->save($cacheKey, $segments);
 
+        if ($_hideSegmentsIfDataNotProfilable
+            && !Request::isCurrentPeriodProfilable()
+        ) {
+            $segments = array_filter($segments, function ($data) {
```

Comment:
wondering the same as with the reports... when changing the date in the UI then the list of segments might not change if the list of segments was loaded earlier for a different date? It may be fine though. I reckon we might always want to show all segments though maybe as in different periods some data may be profilable and then we would want to be able to create segments for it? Bit hard to explain maybe what I mean

Reply:
I suppose it's less of an issue when a site always doesn't use visitorId (profilable) then the reports will be consistent and we do a reload when changing the site anyway. It's only an issue when changing to a date where the behaviour was different but this might be fine although it can cause an issue where it means that certain reports where profilables exist won't be shown maybe. Eg we removed for matomo org cookies a while ago and wondering if we view today, and then go back in date for say 1 year, we wouldn't see some data anymore. It would affect only few data though because the most important reports would still exist and it more impacts only unique visitors etc which would be still shown when going back in time without reloading the page I think.

Reply:
@mattab can you take a look at ^? (that we should show all segments in API output but w/ eg `requiresProfilableData` property instead of just hiding them and filter in the UI)

Reply:
showing all segments in API output and when the currently used segment is not profilable, a warning is displayed